pygments/lexers/graphql.py

```diff
@@ -0,0 +1,205 @@
+from pygments.lexer import RegexLexer, words, include, bygroups
+from pygments.token import Comment, Keyword, Name, Number, Punctuation, String, Text
+
+
+__all__ = ["GraphQLLexer"]
+
+OPERATION_TYPES = ("query", "mutation", "subscription")
+BUILTIN_TYPES = ("Int", "Float", "String", "Boolean", "ID")
+BOOLEAN_VALUES = ("true", "false", "null")
+KEYWORDS = (
+    "type",
+    "schema",
+    "extend",
+    "enum",
+    "scalar",
+    "implements",
+    "interface",
+    "union",
+    "input",
+    "directive",
+    "QUERY",
+    "MUTATION",
+    "SUBSCRIPTION",
+    "FIELD",
+    "FRAGMENT_DEFINITION",
+    "FRAGMENT_SPREAD",
+    "INLINE_FRAGMENT",
+    "SCHEMA",
+    "SCALAR",
+    "OBJECT",
+    "FIELD_DEFINITION",
+    "ARGUMENT_DEFINITION",
+    "INTERFACE",
+    "UNION",
+    "ENUM",
+    "ENUM_VALUE",
+    "INPUT_OBJECT",
+    "INPUT_FIELD_DEFINITION",
+)
+
+
+class GraphQLLexer(RegexLexer):
+
+    name = "GraphQL"
+    aliases = ["graphql"]
+    filenames = ["*.graphql"]
+
+    tokens = {
+        "ignored_tokens": [
+            (r"([ \t]|[\r\n]|\r\n)+", Text),  # Whitespaces
+            (r"#.*$", Comment),
+            (",", Punctuation),  # Insignificant commas
+        ],
+        "value": [
+            include("ignored_tokens"),
+            include("value_single0"),
+        ],
+        # How to avoid state duplication?
+        "value_single0": [
+            include("ignored_tokens"),
+            (
+                r"-?\d+\.\d+[eE][+-]?\d+|-?\d+\.\d+|-?\d+[eE][+-]?\d+",
+                Number.Float,
+                "#pop",
+            ),
+            (r"-?\d+", Number.Integer, "#pop"),
+            (r'"', String, "string0"),
+            (words(BOOLEAN_VALUES, suffix=r"\b"), Name.Builtin, "#pop"),
+            (r"\$[a-zA-Z_]\w*", Name.Variable, "#pop"),
```

Comment:
Is Unicode supported? If so, `[^\W\d]` would be better than `[a-zA-Z_]` (same thing in other places).